packages/flutter_adaptive_scaffold/test/adaptive_scaffold_test.dart

```diff
@@ -152,34 +200,95 @@ class NeverOnBreakpoint extends Breakpoint {
-}) async {
-  await tester.binding.setSurfaceSize(Size(width, 800));
-  return MaterialApp(
-    home: MediaQuery(
-      data: MediaQueryData(size: Size(width, 800)),
-      child: AdaptiveScaffold(
-        drawerBreakpoint: NeverOnBreakpoint(),
-        internalAnimations: animations,
-        smallBreakpoint: TestBreakpoint0(),
-        mediumBreakpoint: TestBreakpoint800(),
-        largeBreakpoint: TestBreakpoint1000(),
-        destinations: const <NavigationDestination>[
-          NavigationDestination(icon: Icon(Icons.inbox), label: 'Inbox'),
-          NavigationDestination(icon: Icon(Icons.article), label: 'Articles'),
-          NavigationDestination(icon: Icon(Icons.chat), label: 'Chat'),
-          NavigationDestination(icon: Icon(Icons.video_call), label: 'Video'),
-        ],
-        smallBody: (_) => Container(color: Colors.red),
-        body: (_) => Container(color: Colors.green),
-        largeBody: (_) => Container(color: Colors.blue),
-        smallSecondaryBody: (_) => Container(color: Colors.red),
-        secondaryBody: (_) => Container(color: Colors.green),
-        largeSecondaryBody: (_) => Container(color: Colors.blue),
-      ),
+class TestScaffold extends StatefulWidget {
+  const TestScaffold({
+    super.key,
+    this.initialIndex = 0,
+    this.isAnimated = true,
+  });
+
+  final int initialIndex;
+  final bool isAnimated;
+
+  static const List<NavigationDestination> destinations =
+      <NavigationDestination>[
+    NavigationDestination(
+      key: Key('Inbox'),
+      icon: Icon(Icons.inbox),
+      label: 'Inbox',
+    ),
+    NavigationDestination(
+      key: Key('Articles'),
+      icon: Icon(Icons.article),
+      label: 'Articles',
+    ),
+    NavigationDestination(
+      key: Key('Chat'),
+      icon: Icon(Icons.chat),
+      label: 'Chat',
     ),
-  );
+  ];
+
+  @override
+  State<TestScaffold> createState() => TestScaffoldState();
+}
+
+class TestScaffoldState extends State<TestScaffold> {
+  late int index = widget.initialIndex;
+
+  @override
+  Widget build(BuildContext context) {
+    return AdaptiveScaffold(
+      selectedIndex: index,
+      onSelectedIndexChange: (int index) {
+        setState(() {
+          this.index = index;
+        });
+      },
+      drawerBreakpoint: NeverOnBreakpoint(),
+      internalAnimations: widget.isAnimated,
+      smallBreakpoint: TestBreakpoint0(),
+      mediumBreakpoint: TestBreakpoint800(),
+      largeBreakpoint: TestBreakpoint1000(),
+      destinations: TestScaffold.destinations,
+      smallBody: (_) => Container(color: Colors.red),
+      body: (_) => Container(color: Colors.green),
+      largeBody: (_) => Container(color: Colors.blue),
+      smallSecondaryBody: (_) => Container(color: Colors.red),
+      secondaryBody: (_) => Container(color: Colors.green),
+      largeSecondaryBody: (_) => Container(color: Colors.blue),
+    );
+  }
+}
+
+enum SimulatedLayout {
+  mobile(width: 400, navSlotKey: Key('bottomNavigation')),
```

Comment:
The previous `scaffold` method logic is wrapped so that it can also inform the program of the expected size. IMO, this helps the readability of the test.